testbed/addons/kit/construct.ts

```diff
@@ -0,0 +1,97 @@
+import cdk = require('@aws-cdk/core')
+import eks = require('@aws-cdk/aws-eks')
+import iam = require('@aws-cdk/aws-iam')
+
+export interface KitProps {
+    cluster: eks.Cluster
+}
+
+export class Kit extends cdk.Construct {
+    constructor(scope: cdk.Construct, id: string, props: KitProps) {
+        super(scope, id)
+        const namespace = "kit"
+        const ns = props.cluster.addManifest('kit-namespace', {
+            apiVersion: 'v1',
+            kind: 'Namespace',
+            metadata: {
+                name: namespace
+            }
+        })
+
+        // Controller Role
+        const sa = props.cluster.addServiceAccount('kit-controller-sa', {
+            name: "kit-controller",
+            namespace: namespace
+        })
+        sa.node.addDependency(ns)
+        sa.role.attachInlinePolicy(new iam.Policy(this, 'kit-controller-policy', {
+            statements: [
+                new iam.PolicyStatement({
+                    resources: ['*'],
+                    actions: [
+                        // Write Operations
+                        "ec2:CreateTags",
+                        "ec2:CreateLaunchTemplate",
+                        "ec2:CreateLaunchTemplateVersion",
+                        "ec2:DeleteLaunchTemplate",
+                        "ec2:RunInstances",
+                        "iam:passRole",
+                        "autoscaling:CreateOrUpdateTags",
+                        "autoscaling:CreateAutoScalingGroup",
+                        "autoscaling:DeleteAutoScalingGroup",
+                        "autoscaling:UpdateAutoScalingGroup",
+                        "autoscaling:SetDesiredCapacity",
+                        //Read Operations
+                        "ec2:DescribeInstances",
+                        "ec2:DescribeLaunchTemplates",
+                        "ec2:DescribeLaunchTemplateVersions",
+                        "ec2:DescribeSubnets",
+                        "ssm:GetParameter",
+                        "autoscaling:DescribeAutoScalingGroups"]
+                }),
+            ],
+        }))
+
+        // Node Role
+        const nodeRole = new iam.Role(this, 'kit-node-role', {
+            assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+            managedPolicies: [
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'),
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'),
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'),
+                iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore')
+            ]
+        })
+
+        props.cluster.awsAuth.addRoleMapping(nodeRole, {
```

Comment:
I don't think we need this, whats the use-case here?

Reply:
I usually use it for debugging purposes by assuming role from CLI outside cluster to verify if this working as expected.

Reply:
I am curious how do you use it, lets maybe sync on a call?